spec/namespaces.md

```diff
@@ -34,14 +39,21 @@ class B {}
 
 The two compilation units contribute to the single global namespace, in this case declaring two classes with the fully qualified names `A` and `B`. Because the two compilation units contribute to the same declaration space, it would have been an error if each contained a declaration of a member with the same name.
 
+A *single_line_namespace* will contribute members corresponding to the *namespace_declaration* it is semantically equivalent to.  See ([Namespace Declarations](#namespace-declarations)) for more details.
+
 ## Namespace declarations
 
 A *namespace_declaration* consists of the keyword `namespace`, followed by a namespace name and body, optionally followed by a semicolon.
+A *single_line_namespace_declaration* consists of the keyword `namespace`, followed by a namespace name, a semicolon and an optional list of directives and type declarations.
 
 ```antlr
 namespace_declaration
     : 'namespace' qualified_identifier namespace_body ';'?
     ;
+    
+single_line_namespace_declaration
+    : 'namespace' qualified_identifier ';' extern_alias_directive* using_directive* type_declaration*
```

Comment:
What's the point of allowing more using directives after the single-line namespace declaration?

I'd be tempted to do:
```
single_line_namespace_declaration
    : 'namespace' qualified_identifier ';' type_declaration*
    ;
```

Reply:
> What's the point of allowing more using directives after the single-line namespace declaration?

It's a very core part of this design.  We want the common file case (which has a single namespace, and usings either outside or inside of it) to always be convertible with no change in semantics.

You hit the nail on the head in your followup comment.  We do not want to punish uesrs who put usings inside (which is also a big style that is commonly seen, and i think stylecop pushed people toward).

Reply:
Sometimes I need usings inside to resolve extensions methods.

https://gist.github.com/ufcpp/f30f4876cffea6a42f1b4a810b92b1ff